### PR TITLE
Update rules_haskell

### PR DESCRIPTION
**Pull request by aherrmann-da**
- A few patches have been upstreamed and can be removed now.
- Updates `rules_nixpkgs` to include https://github.com/tweag/rules_nixpkgs/pull/94

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Add a line to the [release notes](https://github.com/digital-asset/daml/blob/master/unreleased.rst), if appropriate
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: bazel_tools/haskell-ghci-grpc.patch [hunk 1]
```diff
@@ -1,9 +1,66 @@
 On MacOS GHCi fails to load grpc fat_cbits statically due to duplicate symbols.
+diff --git a/haskell/private/path_utils.bzl b/haskell/private/path_utils.bzl
+index 3956164a..037cc95a 100644
+--- a/haskell/private/path_utils.bzl
++++ b/haskell/private/path_utils.bzl
+@@ -135,6 +135,44 @@ def make_library_path(hs, libs, prefix = None):
+ 
+     return join_path_list(hs, set.to_list(r))
+ 
++def symlink_dynamic_library(hs, lib, outdir):
```
Comment (aherrmann-da) [inline bazel_tools/haskell-ghci-grpc.patch]: This function was removed from upstream rules_haskell. We still need it for `fat_cbits`, so it's added to this patch.

File: bazel_tools/ghci-template.sh [hunk 2]
```diff
@@ -4,6 +4,11 @@
 
 ENV_FILE=$1
 ARGS_FILE=$2
-RULES_HASKELL_EXEC_ROOT=$(dirname $(readlink ${BUILD_WORKSPACE_DIRECTORY}/bazel-out))/
+RULES_HASKELL_EXEC_ROOT=bazel-$(basename ${BUILD_WORKSPACE_DIRECTORY})/
```
Comment (cocreature) [inline bazel_tools/ghci-template.sh]: Was this a necessary change or just a simplification?

File: deps.bzl [hunk 3]
```diff
@@ -41,18 +41,18 @@ def daml_deps():
...
-                "@com_github_digital_asset_daml//bazel_tools:haskell-bazel-1.1.0.patch",
-                # Remove once https://github.com/tweag/rules_haskell/pull/1039 is merged.
-                "@com_github_digital_asset_daml//bazel_tools:haskell-cc-wrapper.patch",
-                # Upstream once https://github.com/tweag/rules_haskell/pull/1039 is merged.
-                # Used to work around https://github.com/tweag/rules_haskell/issues/1062.
-                "@com_github_digital_asset_daml//bazel_tools:haskell-cc-wrapper-include-dirs.patch",
-                "@com_github_digital_asset_daml//bazel_tools:haskell-cc-wrapper-globbing.patch",
+                # This is a daml specific patch and not upstreamable.
                 "@com_github_digital_asset_daml//bazel_tools:haskell-windows-extra-libraries.patch",
+                # This is a daml specific patch and not upstreamable.
```
Comment (cocreature) [inline deps.bzl]: Thanks a lot for documenting these!

File: bazel_tools/ghci-template.sh (hunk 2 above)
Comment (aherrmann-da) [reply bazel_tools/ghci-template.sh]: That was a necessary change, see [commit message](https://github.com/digital-asset/daml/pull/3275/commits/e12176a02206dfb553fb9ebe5883c9d0b342d227)
> Cabal package library paths are longer and overflow the command-line
length when calling `cc` on MacOS for template Haskell dependencies in
the IDE tests. Shortening the `RULES_HASKELL_EXEC_ROOT` prefix for each
`(LD_)LIBRARY_PATH` entry works around the issue.

**Pull request merged**